src/optimisation/linear-problem-data-impl/timeSeriesSet.cpp

```diff
@@ -20,16 +44,22 @@ void TimeSeriesSet::add(const std::vector<double>& ts)
     tsSet_.push_back(std::move(ts));
 }
 
-double TimeSeriesSet::getData(unsigned rank, unsigned hour) const
+double TimeSeriesSet::getData(LinearProblemApi::IScenario::TimeSeriesNumber rank,
+                              unsigned hour) const
 {
+    if (rank == 0)
+    {
+        throw Antares::Error::RuntimeError(
+          "Trying to get data set at rank 0. Data sets are indexed starting at 1.");
+    }
     if (tsSet_.empty())
     {
         throw Empty(name());
     }
-
+    rank = rank - 1;
     if (rank > tsSet_.size() - 1)
     {
-        throw RankTooBig(name(), rank);
+        throw RankTooBig(name(), rank + 1);
     }
```

Comment:
Create a temporary, such as `int rankIndexedFrom0 = rankIndexedFrom1 - 1` to make things obvious